src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTemplateEngineFactoryService.cs

```diff
@@ -72,28 +74,20 @@ public override RazorTemplateEngine Create(string projectPath, Action<IRazorEngi
             }
         }
 
-        private Version GetMvcVersion(string projectPath)
+        private ProjectSnapshot FindProject(string directory)
         {
-            var workspace = _services.WorkspaceServices.Workspace;
+            directory = NormalizeDirectoryPath(directory);
 
-            var project = workspace.CurrentSolution.Projects.FirstOrDefault(p =>
+            var projects = _projectManager.Projects;
+            for (var i = 0; i < projects.Count; i++)
             {
-                var directory = Path.GetDirectoryName(p.FilePath);
-                return string.Equals(
-                    NormalizeDirectoryPath(directory),
-                    NormalizeDirectoryPath(projectPath),
-                    StringComparison.OrdinalIgnoreCase);
-            });
-
-            if (project != null)
-            {
-                var compilation = CSharpCompilation.Create(project.AssemblyName).AddReferences(project.MetadataReferences);
-
-                foreach (var identity in compilation.ReferencedAssemblyNames)
+                var project = projects[i];
+                if (project.UnderlyingProject.FilePath != null)
                 {
-                    if (identity.Name == MvcAssemblyName)
+                    Path.GetDirectoryName(project.UnderlyingProject.FilePath);
+                    if (string.Equals(directory, NormalizeDirectoryPath(Path.GetDirectoryName(project.UnderlyingProject.FilePath)), StringComparison.OrdinalIgnoreCase))
```

Comment:
Doing ignore case here might be troublesome once VS for mac pops into the picture.

Reply:
This whole thing is inexact